sdk/python/toolchain/toolchain_test.go

```diff
@@ -194,6 +195,81 @@ func TestAbout(t *testing.T) {
 	}
 }
 
+//nolint:paralleltest // mutates environment variables
+func TestPyenv(t *testing.T) {
```

Comment:
Note we don't have any integration tests for pyenv because it is tricky to have pyenv available in GitHub Actions.

The tests for the pulumi docker containers do test the functionality e2e https://github.com/pulumi/pulumi-docker-containers/pull/232